src/img/tester.py

```diff
@@ -511,28 +511,44 @@ def install_tests(self):
             packages = []
             patterns = []
             host_test_packages = []
-            for name in self.test_packages.keys():
+            pre_check = []
+            skip_already_installed = (
+                'rpm -qi {package} ||'
+            )
+            for name in self.test_packages:
```

Comment:
Why have we gone from name in test_packages.keys(): to test_packages: ??

Reply:
for-loop itself unpacks keys from dict, so calling .keys() is overhead for iterating dict keys